dll/win32/shell32/dialogs/item_prop.cpp

```diff
@@ -0,0 +1,182 @@
+/*
+ * PROJECT:     shell32
+ * LICENSE:     LGPL-2.1+ (https://spdx.org/licenses/LGPL-2.1+)
+ * PURPOSE:     FileSystem PropertySheet implementation
+ * COPYRIGHT:   Copyright 2024 Whindmar Saksit <whindsaks@proton.me>
+ */
+
+#include "precomp.h"
+
+WINE_DEFAULT_DEBUG_CHANNEL(shell);
+
+static HRESULT
+SHELL_GetCaptionFromDataObject(IDataObject *pDO, LPWSTR Buf, UINT cchBuf)
+{
+    HRESULT hr = E_INVALIDARG;
+    if (PIDLIST_ABSOLUTE pidl = SHELL_DataObject_ILCloneFullItem(pDO, 0))
+    {
+        hr = SHGetNameAndFlagsW(pidl, SHGDN_INFOLDER, Buf, cchBuf, NULL);
+        ILFree(pidl);
+    }
+    return hr;
+}
+
+struct ShellPropSheetDialog
+{
+    typedef void (CALLBACK*PFNINITIALIZE)(LPCWSTR InitString, IDataObject *pDO,
+                                          HKEY *hKeys, UINT *cKeys);
+
+    static HRESULT Show(const CLSID *pClsidDefault, IDataObject *pDO,
+                        PFNINITIALIZE InitFunc, LPCWSTR InitString)
+    {
+        HRESULT hr;
+        CRegKeyHandleArray keys;
+        if (InitFunc)
+            InitFunc(InitString, pDO, keys, keys);
+        WCHAR szCaption[MAX_PATH], *pszCaption = NULL;
+        if (SUCCEEDED(SHELL_GetCaptionFromDataObject(pDO, szCaption, _countof(szCaption))))
+            pszCaption = szCaption;
```

Comment:
If the SHELL_GetCaptionFromDataObject failed, pszCaption remains NULL. Can the SHOpenPropSheetW function cope with that?